files/usr/bin/user-flatpak-setup

```diff
@@ -25,36 +14,43 @@ REPO_TITLE=$(yq '.repo-title' $REPO_INFO)
 # Set up per-user Flatpak repository
 if [[ ! $REPO_URL == "null" && ! $REPO_NAME == "null" ]]; then
   echo "Adding remote $REPO_NAME from $REPO_URL"
-  flatpak remote-add --if-not-exists --user $REPO_NAME $REPO_URL
+  flatpak remote-add --if-not-exists --user "$REPO_NAME" "$REPO_URL"
 fi
 
 # Change repository title to configured title, if not null
 if [[ ! $REPO_TITLE == "null" ]]; then
   echo "Setting title $REPO_TITLE for remote $REPO_NAME"
-  flatpak remote-modify --user $REPO_NAME --title="$REPO_TITLE"
+  flatpak remote-modify --user "$REPO_NAME" --title="$REPO_TITLE"
 fi
 
-# Lists of flatpaks
+# Installed flatpaks
 FLATPAK_LIST=$(flatpak list --columns=application)
-INSTALL_LIST=$(cat /etc/flatpak/user/install)
-REMOVE_LIST=$(cat /etc/flatpak/user/remove)
+
+# Flatpak list files
+INSTALL_LIST_FILE="/etc/flatpak/user/install"
+REMOVE_LIST_FILE="/etc/flatpak/user/remove"
 
 # Install flatpaks in list
-if [[ -n $INSTALL_LIST ]]; then
-  if ! flatpak install --user --noninteractive $REPO_NAME ${INSTALL_LIST[@]}; then
-    # Exit on error
-    exit 1
+if [[ -f $INSTALL_LIST_FILE ]]; then
+  if [[ -n $FLATPAK_LIST ]]; then
+    INSTALL_LIST=$(echo "$FLATPAK_LIST" | grep -o -vf "$INSTALL_LIST_FILE")
+  else
+    INSTALL_LIST=$(cat $INSTALL_LIST_FILE)
+  fi
+  if [[ -n $INSTALL_LIST ]]; then
+    if ! flatpak install --user --noninteractive "$REPO_NAME" "${INSTALL_LIST[@]}"; then
+      # Exit on error
+      exit 1
+    fi  
   fi
 fi
 
 # Remove flatpaks in list
-if [[ -n $REMOVE_LIST ]]; then
-  flatpak remove --user --noninteractive $flatpak ${REMOVE_LIST[@]}
+if [[ -f $REMOVE_LIST_FILE ]]; then
```

Comment:
> Would also checking for the existence of the version file like in Bazzite so that this runs once. Otherwise, you could remove a flatpak someone intentionally reinstalled and is currently using.

For this usecase, I intended this to be like this for myself:

OS installs "system" flatpaks, user installs "user" flatpaks (I configured that Gnome Software uses "user" remote by default).

If OS removes "system" flatpak which user needs, than user can install this flatpak with "user" remote. He can backup & restore data from "system" flatpak from $HOME/.var/app as we don't delete that.

But I understand this usecase, especially if user did not configure user remote as default for installation, so it's all good.